File: Python/344. ReverseString.py
#Function to reverse a string
class Solution(object):
    def reverseString(self, s):
        #Taking variables that point to the starting and end position
        i=0
        j=len(s) - 1
        while(i <= j):
            #Swapping the starting and the ending value
            s[i] , s[j] = s[j] , s[i]
            #Incrementing the variable that points to the starting position and decrementing the variable that points to the last position.
            #The loop will continue till the variable pointing the starting value becomes more than the one pointing the ending value.
            #This will assure all the characters in the string are swapped. Hence reversing the string
            i +=1
            j -=1

File: C++/344.ReverseString.cpp
class Solution {
public:
    void reverseString(vector<char>& s) {
        //1st pointer pointing to the start of the string
        int a = 0;
        //2nd pointer poiting to the end of the string
        int b = s.size()-1;

        //iterate over the string till a<=b
        while(a<=b)
        {
            //swap the 2 index values of the string
            swap(s[a], s[b]);
            a++;
            b--;
        }
    }
};

//TC: O(1), SC=O(1)
